src/App.tsx

```diff
@@ -1,61 +1,122 @@
+  ...todo,
+  user: getUser(todo.userId),
+}));
+
+export const App:React.FC = () => {
+  const [visibleTodos, setVisibleTodos] = useState([...todos]);
+  const [title, setTitle] = useState('');
+  const [selectedUser, setSelectedUser] = useState(0);
+  const [errorExistence, setErrorExistence] = useState(false);
+  const [selectError, setSelectError] = useState(false);
+
+  const titleTrim = title.trim().length;
+
+  const handleTitle = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setTitle(event.target.value);
+    setSelectError(false);
+  };
+
+  const handleUserName = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    setSelectedUser(Number(event.target.value));
+    setErrorExistence(false);
+  };
+
+  const handleSubmit = (event: React.ChangeEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    if (!titleTrim) {
+      setSelectError(true);
+    }
+
+    if (!selectedUser) {
+      setErrorExistence(true);
+    }
+
+    if (title.trim().length && selectedUser) {
+      const newTodo = {
+        userId: selectedUser || 0,
+        id: Math.max(...visibleTodos.map(todo => todo.id)) + 1,
+        title: title.trim(),
+        completed: selectError,
+        user: usersFromServer.find(user => user.id === selectedUser) || null,
+      };
+
+      setVisibleTodos([...visibleTodos, newTodo]);
+      setTitle('');
+      setSelectedUser(0);
+      setSelectError(false);
+      setErrorExistence(false);
+    }
+  };
 
-export const App = () => {
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/users" method="POST">
+      <form
+        action="/api/users"
+        method="POST"
+        onSubmit={handleSubmit}
+      >
         <div className="field">
-          <input type="text" data-cy="titleInput" />
-          <span className="error">Please enter a title</span>
+          Title:&nbsp;
+          <input
+            type="text"
+            data-cy="titleInput"
+            name="todoTitle"
+            placeholder="Please enter a title"
+            value={title}
+            onChange={handleTitle}
+          />
+
+          {selectError && (
+            <span className="error">Please enter a title</span>
+          )}
         </div>
 
         <div className="field">
-          <select data-cy="userSelect">
-            <option value="0" disabled>Choose a user</option>
+          User:&nbsp;
+          <select
+            data-cy="userSelect"
+            name="todoUser"
+            value={selectedUser}
+            onChange={handleUserName}
+          >
+            <option value={0} disabled>Choose a user</option>
+            {usersFromServer.map(user => (
+              <option key={user.id} value={user.id}>
+                {user.name}
+              </option>
+            ))}
           </select>
 
-          <span className="error">Please choose a user</span>
+          {errorExistence && (
+            <span className="error">Plese choose a user</span>
```

Comment:
```suggestion
            <span className="error">Please choose a user</span>
```
Probably that's the reason why you can't pass the test